oda_api/token.py

```diff
@@ -62,57 +68,154 @@ def decode_oda_token(token: str, secret_key=None, allow_invalid=False) -> dict:
 
     raise RuntimeError()
 
+
 def decode_oauth2_token(token: str):
     # usually comes in cookies['_oauth2_proxy']
     return json.loads(base64.b64decode(token.split(".")[0]+"=").decode())
 
-#TODO: move to dynaconf
-def discover_token(
+
+def get_token_roles(decoded_token):
+    # extract role(s)
+    roles = None
+    if isinstance(decoded_token['roles'], str):
+        roles = decoded_token['roles'].split(',') if 'roles' in decoded_token else []
```

Comment:
we just check that `decoded_token['roles']` is `str`, it's definitily in there :)